Code/GraphMol/SynthonSpaceSearch/SynthonSet.cpp

```diff
@@ -382,6 +451,69 @@ void SynthonSet::buildSynthonFingerprints(
   }
 }
 
+void SynthonSet::buildAddAndSubtractFPs(
```

Comment:
If I'm reading this correctly (and I have a cold, so I may not be), I think that two searches done using default search parameters could lead to different results. That's sub-optimal.

I think this would be less surprising if it were 100% deterministic, either use a fixed random number seed in this code or have a separate seed parameter for this initialization that defaults to a standard value.

Reply:
It is true that it will give slightly different results on different runs, and I have seen this happen.  I wanted a quick way of sampling the data that wasn't dependent on the input order.  I wonder how fast it would be to select, say, a tenth of each synthon set using the LeaderPicker or some such.  That might give a better assessment of the overall products and thus the add and subtract fingerprints.  I will investigate.

Reply:
I think using a random sample is fine, I would just prefer to have the results be reproducible-by-default, and this isn't